src/qforte/ucc/uccnvqe.py

```diff
@@ -53,6 +54,7 @@ def run(self,
         self._optimizer = optimizer
         self._pool_type = pool_type
         self._noise_factor = noise_factor
+        self._irrep = irrep
```

Comment:
`_irrep` should not be a property of the `Ansatz`. It should be a property of `System`, which any `Ansatz` has. Please remove on all three subclasses and change the population of the operator pool accordingly.

Reply:
I do not necessarily agree with this. The irreducible representation of a particular state is not a property of the system, in our case molecule. A given molecule, at a given geometry, belongs to a given point group, but nothing can be said about an irreducible representation, unless we commit to a given state. The way I see it is as follows. The system class defines our Fermi vacuum, which for closed-shell species translates to a totally symmetric state (I believe that, for the time being, QForte does not work for open-shell molecules, but please correct me if I am wrong). If we want to study states that belong to a different irreducible representation than the RHF Slater determinant, we need to construct an Ansatz that belongs to the irreducible representation that we want. For example, in GAMESS, and I am pretty sure that this is how it is done in Psi4 as well, if one wants to perform an FCI calculation they need to provide the irreducible representation of the target state (if nothing is specified, the default is totally symmetric). This is the reason why `_irrep` should be an attribute of the Ansatz and not the `System`.

Reply:
Right, I was reviewing this a little too quickly. `_irrep_` should definitely not be a property of `System`.

I still do not want `_irrep` as a member of the `Ansatz` class. `_irrep` makes perfect sense for ansatze expressed in terms of fermionic second quantization, but not for an ansatz expressed in terms of Pauli strings. What about having a member called `symmetries` or `states` that is a dict to store all the relevant symmetries? For molecular systems, we can have a key called `_irrep` that hashes to the irrep of the state of interest.

Reply:
I like this idea. Eventually, I could add the information about the Z2 symmetries there as well. I just want to confirm that I understand this correctly, so that I can implement it. Symmetries/states would be a class of its own or an attribute to an existing class? The word "member" confuses me a little :-)

Reply:
Attribute of an existing class.